packages/components/src/components/hds/disclosure-primitive/index.ts

```diff
@@ -33,11 +33,38 @@ export interface HdsDisclosurePrimitiveSignature {
 }
 
 export default class HdsDisclosurePrimitiveComponent extends Component<HdsDisclosurePrimitiveSignature> {
-  @tracked isOpen = this.args.isOpen ?? false;
+  @tracked _isOpen = false;
+  @tracked _isControlled = this.args.isOpen != undefined;
+
+  get isOpen() {
+    if (this._isControlled) {
+      // if the state is controlled from outside, the argument overrides the internal state
+      return this.args.isOpen !== undefined ? this.args.isOpen : this._isOpen;
```

Comment:
I think this could be shortened to:

`return this.args.isOpen ?? this._isOpen`

Reply:
Yes, we can simplify this by using nullish coalescing. Updated as suggested.